pkg/flow/memory.go

```diff
@@ -181,6 +218,15 @@ func (engine *engine) getInstanceMemory(ctx context.Context, inc *ent.InstanceCl
 	im.engine = engine
 	im.in = in
 
+	if in.Edges.Orginator == nil {
```

Comment:
We need to be careful of things like this. Will this upgrade an existing installation gracefully?

Reply:
true, i forgot to remove it.